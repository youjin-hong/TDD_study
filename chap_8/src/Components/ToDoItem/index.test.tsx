import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { ToDoItem } from './index';

describe('<Input />', () => {
  it('renders component correctly', () => {
    const { container } = render(<ToDoItem label="default value" />);

    const todoItem = screen.getByText('default value');
    expect(todoItem).toBeInTheDocument();

    const deleteButton = screen.getByText('삭제');
    expect(deleteButton).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('clicks the delete button', () => {
    const handleClick = vi.fn();
    render(<ToDoItem label="default value" onDelete={handleClick} />);

    const deleteButton = screen.getByText('삭제');
    expect(handleClick).toHaveBeenCalledTimes(0);
    fireEvent.click(deleteButton);
    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});
