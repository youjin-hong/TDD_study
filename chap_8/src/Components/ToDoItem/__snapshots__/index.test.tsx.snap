// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Input /> > renders component correctly 1`] = `
<div>
  <div
    class="sc-fhHczv grQgxZ"
  >
    <div
      class="sc-ggWZvA jahAZR"
    >
      default value
    </div>
    <button
      class="sc-bRKDuR dunOfA"
    >
      <div
        class="sc-hvigdm WvKZY"
      >
        삭제
      </div>
    </button>
  </div>
</div>
`;
